src/components/map/WmtsService.js

```diff
@@ -0,0 +1,167 @@
+          };
+          layer.sourceConfig = ol.source.WMTS.optionsFromCapabilities(
+              getCapabilities, layerOptions);
+          layer.attribution = getCapabilities.ServiceProvider.ProviderName;
+          layer['attributionUrl'] =
+              getCapabilities.ServiceProvider.ProviderSite;
+          layer['capabilitiesUrl'] = getCapabilities.OperationsMetadata
+              .GetCapabilities
+              .DCP
+              .HTTP
+              .Get[0]
+              .href;
+          completeWmtsConfig(layer);
+
+          return layer;
+        };
+
+        this.getLayerConfigFromIdentifier = function(getCapabilities,
+            identifier) {
+          var layerConfig;
+          if (getCapabilities.Contents && getCapabilities.Contents.Layer) {
+            getCapabilities.Contents.Layer.forEach(function(layer) {
+              if (layer.Identifier === identifier) {
+                layerConfig = getLayerConfig(getCapabilities, layer);
+              }
+            });
+          }
+
+          return layerConfig;
+        };
+
+        // Create an ol WMTS layer from GetCapabilities informations
+        this.getOlLayerFromGetCapLayer = function(getCapLayer) {
+          getCapLayer.sourceConfig.attributions = [
+            '<a href="' +
+                getCapLayer.attributionUrl +
+                '" target="new">' +
+                getCapLayer.attribution + '</a>'
+          ];
+          var source = new ol.source.WMTS(getCapLayer.sourceConfig);
+          completeWmtsConfig(getCapLayer);
+          var projection = source.getProjection();
+          var extent;
+          if (projection) {
+            extent = projection.getExtent();
+          } else {
+            extent = gaGlobalOptions.defaultExtent;
+          }
+          var layer = new ol.layer.Tile({
+            id: getCapLayer.id,
+            source: source,
+            extent: gaMapUtils.intersectWithDefaultExtent(extent),
+            preload: gaMapUtils.preload,
+            attribution: getCapLayer.sourceConfig.attribution
+          });
+          gaDefinePropertiesForLayer(layer);
+          layer.useThirdPartyData =
+              gaUrlUtils.isThirdPartyValid(getCapLayer.sourceConfig.urls[0]);
+          layer.label = getCapLayer.Title;
+          layer.url = getCapLayer.attributionUrl;
+          layer.attributions = getCapLayer.attributions;
+          layer.timeEnabled = getCapLayer.timeEnabled;
+          layer.timestamps = getCapLayer.timestamps;
+
+          return layer;
+        };
+
+        // Create a WMTS layer and add it to the map
+        this.addWmtsToMap = function(map, layerOptions, index) {
+          var olLayer = this.getOlLayerFromGetCapLayer(layerOptions);
+          if (index) {
+            map.getLayers().insertAt(index, olLayer);
+          } else {
+            map.addLayer(olLayer);
+          }
+          return olLayer;
+        };
+
+        this.importDimensions = function(formatedDimensions) {
```

Comment:
Yes if we use it . I'm just thinking we only manage Time in our app so the current timestamp of a layer should go in the `layers_timestamp` parmalink param but currently it doesn't because `layers_timestamp`is only activatzed with timeEnabled layer and maybe it's a mistake.

So I say yes for this PR creates serialize and deserialize function but we'll see in the future if we can use the `layers_timestamp` param instead.